Apps/WebClient/src/ClientApp/src/components/NotificationCentreComponent.vue

```diff
@@ -63,10 +69,72 @@ function handleClickNotificationAction(notification: Notification): void {
     } else if (notification.categoryName === bctOdrCategory) {
         router.push({ path: "/services" });
     } else {
-        router.push({ path: notification.actionUrl });
+        const isExternal = isExternalUrl(notification.actionUrl);
+        logger.debug(`is External: ${isExternal}`);
+
+        if (isExternal) {
+            // Open the external url in a new tab/window
+            window.open(notification.actionUrl, "_blank");
+        } else {
+            const internalRoute = stripInternalPath(notification.actionUrl);
+            const resolvedRoute = router.resolve(internalRoute);
+
+            if (resolvedRoute.route.matched.length > 0) {
+                if (route.value.path !== internalRoute) {
+                    router.push({ path: internalRoute });
+                }
+            } else {
+                logger.error(
+                    `Invalid internal link: ${notification.actionUrl}`
+                );
+            }
+        }
     }
 }
 
+function isExternalUrl(url: string): boolean {
+    const currentDomain = window.location.hostname;
+    logger.debug(`Domain: ${currentDomain}`);
+
+    // Create a regular expression to extract the domain from the URL.
+    const domainRegex =
+        /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)/im;
```

Comment:
This is duplicated above and below, it should be moved to a const variable above.